src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildServerTestFixtureBase.cs

```diff
@@ -52,7 +53,21 @@ public void Dispose()
                     args.Add("-w");
                 }
 
-                var exitCode = application.Execute(args.ToArray());
+                var executeTask = Task.Run(() => application.Execute(args.ToArray()));
+
+                if (executeTask == await Task.WhenAny(executeTask, Task.Delay(Timeout.Infinite, cts.Token)))
+                {
+                    // Complete the Task.Delay
+                    cts.Cancel();
+                }
+                else
+                {
+                    var output = writer.ToString();
+                    throw new TimeoutException($"Shutting down the build server at pipe {PipeName} took longer than expected.{Environment.NewLine}Output: {output}.");
```

Comment:
Wondering if instead the code should just output a warning e.g. using an `ITestOutputHelper`. Unfortunately that would require changing all uses of derived fixtures. Thoughts i.e. is this case actually an error worth crashing the test host about❔

Reply:
After this change, the tests fail rather than crash the testhost. I don't know if a warning would be better, but it sounds like this isn't going to be around for much longer. This is a simple fix that seems to me to be strictly better than what we have now.